rust/azure_iot_operations_connector/src/base_connector/managed_azure_device_registry.rs

```diff
@@ -123,11 +123,18 @@ impl DeviceEndpointClientCreationObservation {
                         "Dropping device endpoint create notification: {device_endpoint_ref:?}. Failed to get Device definition after retries: {e}"
                     );
                     // unobserve as cleanup
-                    DeviceEndpointClient::unobserve_device(
-                        &self.connector_context,
-                        &device_endpoint_ref,
-                    )
-                    .await;
+                    // Spawn a new task to prevent a possible cancellation and ensure the deleted
+                    // notification reaches the application.
```

Comment:
```suggestion
                    // Spawn a new task to prevent a possible cancellation and ensure the deleted
                    // notification reaches the application.
```
there's no deletion notification to the application in this flow - we just wait for the next notification since this one was bad